File: api/db/models/consulta.py
from peewee import DateField, ForeignKeyField, CharField, BooleanField

from db.db import BaseModel
from db.models.paciente import Paciente
from db.models.disponibilidade import Disponibilidade

class Consulta(BaseModel):
    status = CharField() # "agendada", "realizada", "cancelada"
    data = DateField()
    disponibilidade = ForeignKeyField(Disponibilidade, backref='consultas')
    presenca = BooleanField(null=True, default=None) # "presente", "ausente"

File: agenda-psicologo-frontend/src/app/class/Consulta.ts
import { Disponibilidade } from "./Disponibilidade";
import { Paciente } from "./Paciente";

export class Consulta {
  id?: number;
  paciente: Paciente;
  status: string;
  data: string;
  disponibilidade: Disponibilidade;
  presenca?: boolean | null;

  constructor(paciente: Paciente, status: string, data: string, disponibilidade: Disponibilidade, presenca?: boolean) {
    this.paciente = paciente;
    this.status = status;
    this.data = data;
    this.disponibilidade = disponibilidade;
    this.presenca = presenca;
  }
}